lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/AzureBatchEvaluatorShimManager.java

```diff
@@ -0,0 +1,339 @@
+  public void onNext(final RemoteMessage<EvaluatorShimProtocol.EvaluatorShimStatusProto> statusMessage) {
+
+    EvaluatorShimProtocol.EvaluatorShimStatusProto message = statusMessage.getMessage();
+    String containerId = message.getContainerId();
+    String remoteId = message.getRemoteIdentifier();
+
+    LOG.log(Level.INFO, "Got a status message from evaluator shim = {0} with containerId = {1}.",
+        new String[] {remoteId, containerId});
+
+    synchronized (this.outstandingResourceRequests) {
+      ResourceRequestEvent resourceRequestEvent = AzureBatchEvaluatorShimManager.this
+          .outstandingResourceRequests
+          .get(containerId);
+
+      if (resourceRequestEvent == null) {
+        LOG.log(Level.WARNING, "Received an evaluator shim status message from an unknown "
+            + "evaluator shim. Container id = {0}, remote id = {1}.", new String[] {containerId, remoteId});
+      } else {
+        LOG.log(Level.FINEST, "Notifying REEF of a new node: {0}", remoteId);
+        this.reefEventHandlers.onNodeDescriptor(NodeDescriptorEventImpl.newBuilder()
+            .setIdentifier(RemoteIdentifierParser.parseNodeId(remoteId))
+            .setHostName(RemoteIdentifierParser.parseIp(remoteId))
+            .setPort(RemoteIdentifierParser.parsePort(remoteId))
+            .setMemorySize(resourceRequestEvent.getMemorySize().get())
+            .build());
+
+        LOG.log(Level.FINEST, "Firing a new ResourceAllocationEvent for remoteId = {0}.", remoteId);
+        this.reefEventHandlers.onResourceAllocation(
+            ResourceEventImpl.newAllocationBuilder()
+                .setIdentifier(containerId)
+                .setNodeId(RemoteIdentifierParser.parseNodeId(remoteId))
+                .setResourceMemory(resourceRequestEvent.getMemorySize().get())
+                .setVirtualCores(resourceRequestEvent.getVirtualCores().get())
+                .setRuntimeName(RuntimeIdentifier.RUNTIME_NAME)
+                .build());
+      }
+
+      this.outstandingResourceRequests.remove(containerId);
+    }
+
+    this.activeResources.put(containerId, remoteId);
+    this.updateRuntimeStatus();
+  }
+
+  public void onResourceLaunched(final ResourceLaunchEvent resourceLaunchEvent,
+                                 final String command,
+                                 final String evaluatorConfigurationString) {
+
+    String resourceRemoteId = this.activeResources.get(resourceLaunchEvent.getIdentifier());
+
+    EventHandler<EvaluatorShimProtocol.EvaluatorShimControlProto> handler =
+        AzureBatchEvaluatorShimManager.this.remoteManager.getHandler(resourceRemoteId,
+            EvaluatorShimProtocol.EvaluatorShimControlProto.class);
+
+    LOG.log(Level.INFO, "Sending a command to the Evaluator shim to start the evaluator.");
+    handler.onNext(
+        EvaluatorShimProtocol.EvaluatorShimControlProto
+            .newBuilder()
+            .setCommand(EvaluatorShimProtocol.EvaluatorShimCommand.EVALUATOR_LAUNCH)
```

Comment:
Should be "LAUNCH_EVALUATOR" since you are using it as a launch request and not a launch notification.

Reply:
changed.